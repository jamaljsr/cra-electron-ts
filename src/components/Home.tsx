import React from 'react';
import { Card, Button } from 'antd';

const Home = () => {
  return (
    <div>
      <Card title="Welcome to React App">
        <p>Let's get you started...</p>
        <p>
          <Button type="primary">Counter</Button>
        </p>
      </Card>
    </div>
  );
};

export default Home;
