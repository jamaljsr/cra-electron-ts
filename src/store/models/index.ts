import { AnyAction } from 'redux';
import { connectRouter, RouterState } from 'connected-react-router';
import { History } from 'history';
import { reducer, Reducer } from 'easy-peasy';
import counterModel, { CounterModel } from './counter';

export interface RootModel {
  router: Reducer<RouterState, AnyAction>;
  counter: CounterModel;
}

export const createModel = (history: History<any>): RootModel => {
  const rootModel: RootModel = {
    router: reducer(connectRouter(history) as any),
    counter: counterModel,
  };
  return rootModel;
};
