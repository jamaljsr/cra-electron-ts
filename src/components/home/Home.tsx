import React, { useState, useEffect } from 'react';
import { Card, Button, Alert } from 'antd';
import { Link } from 'react-router-dom';
import { useTranslation, Trans } from 'react-i18next';
import { COUNTER } from 'components/Routes';
import { info } from 'electron-log';

const Home = () => {
  useEffect(() => info('Rendering Home component'), []);
  const { t } = useTranslation();

  const [showAlert, setShowAlert] = useState(false);
  const handleClickMe = () => setShowAlert(true);

  return (
    <div>
      {showAlert && (
        <Alert
          message={t('cmps.home.success-text')}
          type="success"
          showIcon
          data-tid="success"
        />
      )}
      <Card title={t('cmps.home.card-title')}>
        <p>{t('cmps.home.card-description')}</p>
        <p>
          <Trans i18nKey="cmps.home.play">
            Play with the{' '}
            <Link to={COUNTER} data-tid="counter-link">
              Counter
            </Link>{' '}
            thing
          </Trans>
        </p>
        <p>
          <Button type="primary" data-tid="me-btn" onClick={handleClickMe}>
            {t('cmps.home.me-btn')}
          </Button>
        </p>
      </Card>
    </div>
  );
};

export default Home;
