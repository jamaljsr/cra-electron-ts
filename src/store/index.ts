import { routerMiddleware } from 'connected-react-router';
import { createStore, createTypedHooks } from 'easy-peasy';
import { createHashHistory } from 'history';
import { createLogger } from 'redux-logger';
import { createModel, RootModel } from './models';

export const history = createHashHistory();

export const createReduxStore = () => {
  // Redux store Configuration
  const middleware = [];

  // Skip redux logs in console during the tests
  if (process.env.NODE_ENV !== 'test') {
    // Logging Middleware
    const logger = createLogger({
      level: 'info',
      collapsed: true,
    });

    middleware.push(logger);
  }

  // Router Middleware
  const router = routerMiddleware(history);
  middleware.push(router);

  const models = createModel(history);

  // create easy-peasy store
  return createStore(models, {
    middleware,
  });
};

const store = createReduxStore();

// export hooks directly from the store to get proper type inference
const typedHooks = createTypedHooks<RootModel>();
export const { useStoreActions, useStoreDispatch, useStoreState } = typedHooks;

// enable hot reload for models
if (process.env.NODE_ENV === 'development') {
  if ((module as any).hot) {
    (module as any).hot.accept('./models', () => {
      const models = createModel(history);
      store.reconfigure(models);
    });
  }
}

export default store;
