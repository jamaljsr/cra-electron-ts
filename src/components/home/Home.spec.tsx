import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { StoreProvider } from 'easy-peasy';
import { ConnectedRouter } from 'connected-react-router';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import { createReduxStore } from 'store';
import Home from './Home';

describe('Home component', () => {
  const renderComponent = () => {
    const store = createReduxStore();
    const app = (
      <StoreProvider store={store}>
        <Provider store={store as any}>
          <ConnectedRouter history={createMemoryHistory()}>
            <Home />
          </ConnectedRouter>
        </Provider>
      </StoreProvider>
    );

    return render(app);
  };

  it('should contain a "Click Me!" button', () => {
    const { getByTestId } = renderComponent();
    expect(getByTestId('me-btn')).toHaveTextContent('cmps.home.me-btn');
  });

  it('should contain a link to Counter page', () => {
    const { getByTestId } = renderComponent();
    expect(getByTestId('counter-link')).toHaveTextContent('Counter');
  });

  it('should not show alert message', () => {
    const { queryByTestId } = renderComponent();
    expect(queryByTestId('success')).toBeFalsy();
  });

  it('should show alert after button is clicked', () => {
    const { getByTestId } = renderComponent();
    const btn = getByTestId('me-btn');
    fireEvent.click(btn);
    expect(getByTestId('success')).toHaveTextContent('cmps.home.success-text');
  });
});
