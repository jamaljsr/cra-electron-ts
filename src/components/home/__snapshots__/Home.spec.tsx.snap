// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render the default state of the ui correctly 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Welcome to React App
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <p>
        Let's get you started...
      </p>
      <p>
        Play with the
         
        <a
          data-tid="counter-link"
          href="/counter"
        >
          Counter
        </a>
      </p>
      <p>
        <button
          class="ant-btn"
          data-tid="me-btn"
          type="button"
        >
          <span>
            Click Me!
          </span>
        </button>
      </p>
    </div>
  </div>
</div>
`;
